source/blender/makesdna/DNA_python_component_types.h

```diff
@@ -39,6 +57,28 @@ typedef struct PythonComponentProperty {
   int itemval;
   float vec[4];
   ListBase enumval;
+  struct bAction *action;
+  struct bArmature *armature;
+  struct Camera *camera;
+  struct Collection *collection;
+  struct Curve *curve;
+  struct Image *image;
+  struct Key *key;
+  struct Library *library;
+  struct Light *light;
+  struct Material *material;
+  struct Mesh *mesh;
+  struct MovieClip *movie_clip;
+  struct bNodeTree *node_tree;
+  struct Object *object;
+  struct ParticleSettings *particle_settings;
+  struct bSound *sound;
+  struct Speaker *speaker;
+  struct Text *text;
+  struct Tex *texture;
+  struct VFont *vector_font;
+  struct Volume *volume;
+  struct World *world;
```

Comment:
I tried to use a single `void *pointer` field instead but failed to make it work that way. I don't really know C/C++, so I suspect there might be a way to do it that I missed.